src/main/java/org/cryptomator/cryptofs/health/shortened/ShortenedNamesCheck.java

```diff
@@ -103,6 +115,36 @@ void checkShortenedName(Path dir) throws IOException {
 			}
 		}
 
+
+		/**
+		 * Checks if the string stored inside the name file is base64url encoded
+		 *
+		 * <em>visible for testing</em>
+		 *
+		 * @return {@link NameEncoding} indicating if it is valid, invalid or affected by https://github.com/cryptomator/cryptofs/issues/121
+		 */
+		NameEncoding isValidEncodedName(String toAnalyse) {
+			int firstNullByte = toAnalyse.indexOf('\0');
+			var beforeFirstNull = firstNullByte == -1 ? toAnalyse : toAnalyse.substring(0, firstNullByte);
+			var afterFirstNull = firstNullByte == -1 ? "" : toAnalyse.substring(firstNullByte);
+
+			boolean isFirstBase64 = BASE64URL.canDecode(beforeFirstNull);
```

Comment:
doesn't the "long file name" contain a file extension? iirc, it does. and `.` is not a valid base64url char.

Reply:
Fixed in db83548